app/controllers/concerns/flashable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Flashable
+  extend ActiveSupport::Concern
+
+  def message_for_notice(invalid_records)
+    msg = "Uploading #{upload_params[:csv_type]} succeeded"
+    msg + (invalid_records.present? ? ' with warnings:' : '.')
+  end
```

Comment:
Flashable is a poorly named mixin for formatting the error messages. I've been using flash.notice for success, or success with warnings, and flash.alert for errors. If there is only a notice it goes into flash.notice, if there are errors and notices they go into flash.alert. I could use flash.warn for success uploading.

As an aside, I looked into rolling back uploads on validation errors, but there is no way for us to easily differentiate between a valid error or an error caused by a free-text row embedded amongst data (see eight_key.csv for example). Unless we require VA Edu users to clean up the CSVs before using, we should accept the upload but display the errors.